src/sql/workbench/contrib/notebook/browser/callout.ts

```diff
@@ -0,0 +1,275 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { attachButtonStyler } from 'sql/platform/theme/common/styler';
+import { Modal } from 'sql/workbench/browser/modal/modal';
+import * as TelemetryKeys from 'sql/platform/telemetry/common/telemetryKeys';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { localize } from 'vs/nls';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import * as DOM from 'vs/base/browser/dom';
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { ITextResourcePropertiesService } from 'vs/editor/common/services/textResourceConfigurationService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { attachModalDialogStyler } from 'sql/workbench/common/styler';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { Deferred } from 'sql/base/common/promise';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox';
+import * as styler from 'vs/platform/theme/common/styler';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { Checkbox } from 'sql/base/browser/ui/checkbox/checkbox';
+import { RadioButton } from 'sql/base/browser/ui/radioButton/radioButton';
+
+export type CalloutStyle = 'LINK' | 'IMAGE' | 'TABLE';
+
+export interface ICalloutOptions {
+	insertTtitle?: string,
+	calloutStyle?: CalloutStyle,
+	insertMarkup?: string,
+	imagePath?: string,
+	embedImage?: boolean
+}
+
+export class Callout extends Modal {
```

Comment:
I'd like to see this thing be a common component instead of under Notebook stuff - where the base Callout class would have the common implementation and then users could extend it like they do with Modal to create specific types of callout dialogs. I forsee this class being something we reuse a lot so that'll save a lot of duplicated code

Reply:
Can you please direct me to a suitable location for this code?

Reply:
Same folder as modal should be fine I think - after all it's just another type of modal dialog - https://github.com/microsoft/azuredatastudio/blob/2d182fcd0318faf56f3033f8cb771f4de271abc3/src/sql/workbench/browser/modal

Reply:
Renamed the class to CalloutDialog. Moved to specified location.